### PR TITLE
docs(server): new docs site for ARFlow server

### PR DESCRIPTION
**Pull request by FelixNgFender**
I replaced the `static.yml` CI action and the `bash` scripts used to build & copy the docs to the static website. We now run all of that on a GitHub runner with a new `website.yml` action.

Added docs for Protobuf schema, 1 action item left here though: the old `message` field in `RegisterResponse` has been changed in the Protobuf schema, but the change has not been reflected in the generated data access layer for the C# client.

```proto
message RegisterResponse {
    string uid = 1; // previously named `message`
}
```

Centralized all remaining `bash` scripts in `python/` to Poetry scripts inside `pyproject.toml`.

Added docs for important packages & tools used in the server.

### CODE REVIEW

**Comment by YiqinZhao** [comment]
Thanks for the contribution, a few thoughts of mine:

- Why is the `message` field changed to `uid`? To align with `DataFrameRequest`?
- I don't think centralizing all bash script code to the `pyproject.toml` file is a good practice. The Python side code will be published to pypi as both a package and a CLI tool. The command defined in the `pyproject.toml` file will be exposed as CLI subcommand, for example: `arflow serve`. The scripts for executing example code shouldn't be added to there.

File: .vscode/settings.json [hunk 1]
```diff
@@ -24,4 +24,7 @@
         "thecakelab",
         "Xihe"
     ],
+    "conventionalCommits.scopes": [
```
Comment (YiqinZhao) [inline .vscode/settings.json:27]: What's the purpose of this setting?

File: protos/arflow/service.proto [hunk 2]
```diff
@@ -55,7 +59,7 @@ message RegisterRequest {
 }
 
 message RegisterResponse {
-    string message = 1;
+    string uid = 1;
```
Comment (YiqinZhao) [inline protos/arflow/service.proto:62]: Why do we want to change the `message` field to `uid`? To align with other messages?

File: python/pyproject.toml [hunk 3]
```diff
@@ -32,4 +32,7 @@ requires = ["poetry-core"]
 build-backend = "poetry.core.masonry.api"
 
 [tool.poetry.scripts]
-serve = "arflow.main:serve"
+serve = "arflow.serve:serve"
+docs = "arflow.docs:make_docs"
+simple-server = "examples.simple_server:main"
+simple-replay = "examples.simple_replay:main"
```
Comment (YiqinZhao) [inline python/pyproject.toml:38]: I don't think centralizing all bash script code to the pyproject.toml file is a good practice. The Python side code will be published to pypi as both a package and a CLI tool. The command defined in the pyproject.toml file will be exposed as CLI subcommand, for example: arflow serve. The scripts for executing example code shouldn't be added to there.

File: .vscode/settings.json (hunk 1 above)
Comment (FelixNgFender) [reply .vscode/settings.json:27]: https://www.conventionalcommits.org/en/v1.0.0/

It's just a setting for Conventional Commit VSCode extension to pick up scopes defined in the workspace

**Review by YiqinZhao — COMMENTED** [review]
Added more comments

File: protos/arflow/service.proto (hunk 2 above)
Comment (FelixNgFender) [reply protos/arflow/service.proto:62]: Yes, that's what I'm thinking

File: python/pyproject.toml (hunk 3 above)
Comment (FelixNgFender) [reply python/pyproject.toml:38]: I see. Will revert this

File: .vscode/settings.json (hunk 1 above)
Comment (YiqinZhao) [reply .vscode/settings.json:27]: Interesting. Do you find it helpful in your typical git workflow?

File: protos/arflow/service.proto (hunk 2 above)
Comment (YiqinZhao) [reply protos/arflow/service.proto:62]: I see. In that sense, we probably want to change the `message` field in `DataFrameResponse` to `uid` as well.

File: .vscode/settings.json (hunk 1 above)
Comment (FelixNgFender) [reply .vscode/settings.json:27]: I just find it more useful for reviewers as they can narrow down the scope of what the commit's changing

Comment (YiqinZhao) [reply .vscode/settings.json:27]: Sounds good. Let me see if we can make this tool part of the standard workflow for this project.

File: protos/arflow/service.proto (hunk 2 above)
Comment (FelixNgFender) [reply protos/arflow/service.proto:62]: Looking into the code, it seems like the only use of `DataFrameResponse` is for the server to ACK the client that a data frame has been received. I think we should keep it as is

```python
class ARFlowService(...):
    
    def data_frame(...):
        ...
        return service_pb2.DataFrameResponse(message="OK")
```

Comment (YiqinZhao) [reply protos/arflow/service.proto:62]: Oh, I realized the `uid` actually means session generated by the registration request. While this message field can mean anything. Sure, we can keep the message field in the `DataFrameResponse` untouched.

**Comment by YiqinZhao** [comment]
The changes look good now. Will merging this PR break the server? Because the `uid` field updates are not reflected on the Unity side.

**Review by YiqinZhao — APPROVED** [review]
LGTM

**Pull request merged**